bundles/org.openhab.binding.bloomsky/src/main/java/org/openhab/binding/bloomsky/internal/connection/BloomSkyConnectApi.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bloomsky.internal.connection;
+
+import static java.util.stream.Collectors.joining;
+import static org.openhab.binding.bloomsky.internal.BloomSkyBindingConstants.*;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.bloomsky.internal.dto.BloomSkyJsonSensorData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link BloomSkyConnectAPI} allows access to the BloomSky rest API by the handlers.
+ *
+ * @author Dave J Schoepel - Initial contribution
+ */
+@NonNullByDefault
+public class BloomSkyConnectApi extends BloomSkyApi {
+
+    private final Logger logger = LoggerFactory.getLogger(BloomSkyConnectApi.class);
+
+    /**
+     * Constructor for handler connections to the BloomSky API
+     *
+     * @param httpClient common client to use for BloomSky rest API GET requests
+     */
+    public BloomSkyConnectApi(HttpClient httpClient) {
+        super(httpClient);
+    }
+
+    /**
+     * Method overridden from {@link BloomSkyApi} to retrieve the base URL for the BloomSky API.
+     *
+     * @return Base URL to use for the BloomSky API
+     */
+    @Override
+    protected String getBaseUrl() {
+        return BASE_API_URL;
```

Comment:
I think `BASE_API_URL` could be narrowed to this class, no ?

Reply:
Yes, thanks for the suggestion.